tests/components/hue_ble/test_config_flow.py

```diff
@@ -0,0 +1,291 @@
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {}
+    with (
+        patch(
+            "homeassistant.components.hue_ble.config_flow.async_ble_device_from_address",
+            return_value=return_device_result,
+        ),
+        patch(
+            "homeassistant.components.hue_ble.config_flow.async_scanner_count",
+            return_value=scanner_count_result,
+        ),
+        patch(
+            "homeassistant.components.hue_ble.config_flow.HueBleLight.connect",
+            return_value=connect_result,
+        ),
+        patch(
+            "homeassistant.components.hue_ble.config_flow.HueBleLight.authenticated",
+            return_value=authenticated_result,
+            new_callable=PropertyMock,
+        ),
+        patch(
+            "homeassistant.components.hue_ble.config_flow.HueBleLight.connected",
+            return_value=connected_result,
+            new_callable=PropertyMock,
+        ),
+        patch(
+            "homeassistant.components.hue_ble.config_flow.HueBleLight.poll_state",
+            return_value=poll_state_result,
+        ),
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_NAME: TEST_DEVICE_NAME, CONF_MAC: TEST_DEVICE_MAC},
+        )
+        await hass.async_block_till_done()
+
+    if expected_error is not None:
+        assert result["type"] == FlowResultType.FORM
+        assert result["errors"] == {"base": expected_error}
+
+        with patch(
+            "homeassistant.components.hue_ble.config_flow.validate_input",
+            return_value=True,
+        ):
+            result = await hass.config_entries.flow.async_configure(
+                result["flow_id"],
+                {CONF_NAME: TEST_DEVICE_NAME, CONF_MAC: TEST_DEVICE_MAC},
+            )
+            await hass.async_block_till_done()
+
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == TEST_DEVICE_NAME
+    assert result["result"].unique_id == dr.format_mac(TEST_DEVICE_MAC)
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_unknown(hass: HomeAssistant, mock_setup_entry: AsyncMock) -> None:
+    """Test form when unknown error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.hue_ble.config_flow.validate_input",
+        side_effect=Exception,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_NAME: TEST_DEVICE_NAME, CONF_MAC: TEST_DEVICE_MAC},
+        )
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {"base": "unknown"}
+
+    with patch(
+        "homeassistant.components.hue_ble.config_flow.validate_input",
```

Comment:
I could but then I would have to use the same messy parameterization to drive it like for the form test, and since it already has 100% coverage I figured this was cleaner.

Reply:
then it's totally fine for me 👍